jscomp/runtime/js_typed_array.ml

```diff
@@ -373,3 +373,77 @@ module Float64Array = struct
 end
 module Float64_array = Float64Array
 [@ocaml.deprecated "use `Float64Array` instead"]
+
+module DataView = struct
+  (** The DataView view provides a low-level interface for reading and writing
+      multiple number types in an ArrayBuffer irrespective of the platform's endianness.
+
+    @see <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView> MDN
+  *)
+
+  type t
+
+  external make : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBuffer : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBufferOffset : array_buffer -> int -> t = "DataView" [@@bs.new]
+  external fromBufferRange : array_buffer -> offset:int -> length:int -> t = "DataView" [@@bs.new]
+
+  external buffer : t -> array_buffer = "" [@@bs.get]
+  external byteLength : t -> int = "" [@@bs.get]
+  external byteOffset : t -> int = "" [@@bs.get]
+
+  external getInt8 : t -> int -> int = "" [@@bs.send]
+  external getUint8 : t -> int -> int = "" [@@bs.send]
+
+  external getInt16 : t -> int -> littleEndian:Js.boolean -> int = "" [@@bs.send]
+  external getInt16BigEndian : t -> int -> (_ [@bs.as 0]) -> int = "" [@@bs.send]
```

Comment:
I would like to put `true`/`false` here when it become supported. Pragmatically, interpreting `0` as falsey value, and `1` as truthy is carved in the stone in JS (which pisses me off, only `false` and `null` (and `undefined`!) must be falsey; and preferably `null` and `undefined` not exist...; but we have what we have)